connector/mqtt/latency-pub-test-prometheus/app/LatencyStore.js

```diff
@@ -0,0 +1,111 @@
+const ss = require('simple-statistics');
+const { logger } = require('@dojot/dojot-module-logger');
+
+const TAG = { filename: 'LatencyStore' };
+
+/** Class to store Latencys and calculate statistics * */
+class LatencyStore {
+  /**
+     *
+     */
+  constructor() {
+    logger.debug('Init LatencyStore...', TAG);
+
+    this.latencies = [];
+    this.max = null;
+    this.min = null;
+  }
+
+  /**
+     * Retrieve array of latencies
+     *
+     * @return {Array} All latencies
+     */
+  getLatencies() {
+    return this.latencies;
+  }
+
+  /**
+     * Clean all store info
+     */
+  cleanLatencies() {
+    logger.debug(`Clean all Latencies, the Latencies before clean: ${this.getLatencies()}`, TAG);
+
+    this.latencies = [];
+    this.max = null;
+    this.min = null;
+  }
+
+  /**
+     * Add a new latency to store
+     *
+     * @param {number} latency in ms
+     */
+  addLatency(latency) {
+    this.latencies.push(latency);
+
+    this.updateMax(latency);
+    this.updateMin(latency);
+  }
+
+  /**
+     * Update the attrs of classe min if param is less then current
+     *
+     * @param {number} Latency
+     */
+  updateMin(latency) {
+    this.min = this.min === null || this.min > latency ? latency : this.min;
+  }
+
+  /**
+     * Update the attrs of classe max if param is more then current
+     *
+     * @param {number} Latency
+     */
+  updateMax(latency) {
+    this.max = this.max === null || this.max < latency ? latency : this.max;
```

Comment:
if this.max is a number, why we check for null?

Reply:
Because I would like to expose the metrics only when I receive a number and zero is a number.